### PR TITLE
style: HTMLElement casting style

### PR DESCRIPTION
**Pull request by bvandrc**
Better to use say, `elem.closest<HTMLInputElement>("input")` than `elem.closest("input") as HTMLInputElement`

### CODE REVIEW

File: packages/core/src/common/utils/domUtils.ts [hunk 1]
```diff
@@ -63,12 +63,8 @@ export function elementIsTextInput(elem: HTMLElement) {
...
  */
 export function getActiveElement(element?: HTMLElement | null, options?: GetRootNodeOptions) {
-    if (element == null) {
-        return document.activeElement;
-    }
-
-    const rootNode = (element.getRootNode(options) ?? document) as DocumentOrShadowRoot & Node;
-    return rootNode.activeElement;
+    const rootNode = (element?.getRootNode(options) ?? document) as DocumentOrShadowRoot & Node;
+    return rootNode.activeElement as HTMLElement | null;
```
Comment (ggdouglas) [inline packages/core/src/common/utils/domUtils.ts]: Can we use a double assertion here to specify the type of `rootNode`?
```ts
export function getActiveElement(element?: HTMLElement | null, options?: GetRootNodeOptions): HTMLElement | null {
    const rootNode = (element?.getRootNode(options) ?? document) as unknown as DocumentOrShadowRoot;
    const activeElement = rootNode.activeElement;
    return activeElement instanceof HTMLElement ? activeElement : null;
}
```
I think this makes the uncertainty of the type conversion a bit more explicit/clear.

File: packages/docs-theme/src/components/documentation.tsx [hunk 2]
```diff
@@ -395,18 +395,18 @@ export class Documentation extends React.PureComponent<DocumentationProps, Docum
...
  */
 function getScrolledReference(offset: number, scrollContainer: HTMLElement = document.documentElement) {
-    const headings = Array.from(scrollContainer.querySelectorAll(".docs-title"));
+    const headings = Array.from(scrollContainer.querySelectorAll<HTMLElement>(".docs-title"));
     while (headings.length > 0) {
         // iterating in reverse order (popping from end / bottom of page)
         // so the first element below the threshold is the one we want.
-        const element = headings.pop() as HTMLElement;
+        const element = headings.pop()!;
         if (element.offsetTop < scrollContainer.scrollTop + offset) {
```
Comment (ggdouglas) [inline packages/docs-theme/src/components/documentation.tsx]: Can we use a null check vs. a non-null assertion here?
```suggestion
        if (element != null && element.offsetTop < scrollContainer.scrollTop + offset) {
```

File: packages/core/src/common/utils/domUtils.ts (hunk 1 above)
Comment (bvandrc) [reply packages/core/src/common/utils/domUtils.ts]: I like that!

File: packages/docs-theme/src/components/documentation.tsx (hunk 2 above)
Comment (bvandrc) [reply packages/docs-theme/src/components/documentation.tsx]: Yeah that's much better.

**Review by ggdouglas — APPROVED** [review]
Thanks for cleaning these types up! 🙂

**Pull request merged**